### PR TITLE
Fix or suppress LGTM findings

### PR DESCRIPTION
**Pull request by artem-smotrakov**
Motivation:

LGTM reports multiple issues. They need to be triaged,
and real ones should be fixed.

Modifications:
- Fixed multiple issues reported by LGTM, such as redundant conditions, resource leaks, typos, possible integer overflows.
- Suppressed false-positives.
- Added a few testcases.

Result:

Fixed several possible issues, get rid of false alarms in the LGTM report.

https://lgtm.com/projects/g/netty/netty/?mode=list&tag=

The suppressed warnings don't look like real issues after you have a closed look at the code around. However, they may still be fixed if we don't want to have many suppression comments, of if we'd like to be paranoids. Please let me know.

### CODE REVIEW

**Comment by artem-smotrakov** [comment]
@chrisvest @normanmaurer What do you think about enabling LGTM checks for pull requests?

https://lgtm.com/projects/g/netty/netty/ci/

That would help to detect possible issues earlier. Also, if the checks were enabled, we would see if the findings get fixed in this pull request.

Another option is to enable GitHub scan jobs for pull requests via GitHub Actions, please see #10669.

File: transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollEventArray.java [hunk 1]
```diff
@@ -109,7 +109,8 @@ int fd(int index) {
 
     private int getInt(int index, int offset) {
         if (PlatformDependent.hasUnsafe()) {
-            return PlatformDependent.getInt(memoryAddress + index * EPOLL_EVENT_SIZE + offset);
+            long n = index * EPOLL_EVENT_SIZE;
+            return PlatformDependent.getInt(memoryAddress + n + offset);
```
Comment (normanmaurer) [inline transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollEventArray.java:113]: why was this changed ?

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Comment by hyperxpro** [comment]
Use SupressWarnings, not Comments.

**Comment by artem-smotrakov** [comment]
> Use SupressWarnings, not Comments.

I like `SuppressWarnings` more. Unfortunately, there are several reasons why I used comments:

- The `SuppressWarnings` annotation may not be applied to a specific line. It has to be applied at least to a method. That doesn't look too good because that would hide other similar issues in the method that may be introduced later. As a result, the new issues may be missed.
- Unfortunately, it looks like that LGTM dons't always recognize the `SuppressWarnings` annotation, and as a result, still shows issues that are supposed to be suppressed. See for example here:

https://github.com/netty/netty/pull/10689/files#diff-046253ecd692ecc30df0d143400042126249830bb1817b97f68574d57f8e1aeaL35-L43

I added the annotation last week but the finding are still reported by LGTM. Maybe applying the annotation to `initialValue()` would help.

File: transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollEventArray.java (hunk 1 above)
Comment (artem-smotrakov) [reply transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollEventArray.java:113]: LGTM reports a potential integer overflow for `index * EPOL_EVENT_SIZE`:

https://lgtm.com/projects/g/netty/netty/snapshot/46cad8a5db67884faf2f41f99ed3a74ffa3735f9/files/transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollEventArray.java?sort=name&dir=ASC&mode=heatmap#x3b93cf5f22ba2792:1

I should have convert `index` and the constant to `long` as well. Now it's fixed.

File: buffer/src/main/java/io/netty/buffer/ByteBufOutputStream.java [hunk 2]
```diff
@@ -133,7 +134,8 @@ public void writeShort(int v) throws IOException {
     public void writeUTF(String s) throws IOException {
         DataOutputStream out = utf8out;
         if (out == null) {
-            utf8out = out = new DataOutputStream(this);
+            // Suppress a warning since the stream is closed in the close() method
+            utf8out = out = new DataOutputStream(this); // lgtm[java/output-resource-leak]
```
Comment (normanmaurer) [inline buffer/src/main/java/io/netty/buffer/ByteBufOutputStream.java:138]: I think we should throw if `closed` is true as otherwise we may create a new instance that is never closed again .

File: buffer/src/main/java/io/netty/buffer/ByteBufOutputStream.java [hunk 3]
```diff
@@ -144,4 +146,17 @@ public void writeUTF(String s) throws IOException {
...
+    @Override
+    public void close() throws IOException {
+        if (closed) {
+            return;
+        }
+        closed = true;
+
+        super.close();
+        if (utf8out != null) {
+            utf8out.close();
```
Comment (normanmaurer) [inline buffer/src/main/java/io/netty/buffer/ByteBufOutputStream.java:159]: move this in a finally block ?

**Review by normanmaurer — CHANGES_REQUESTED** [review]

**Review by normanmaurer — APPROVED** [review]

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Comment by artem-smotrakov** [comment]
One of the test jobs failed with the following message:

```
14:09:23 [ERROR] io.netty.util.internal.NettyBlockHoundIntegrationTest  Time elapsed: 2.623 s  <<< ERROR!
14:09:23 java.lang.IllegalStateException: 
14:09:23 The instrumentation have failed.
14:09:23 It looks like you're running on JDK 13+.
14:09:23 You need to add '-XX:+AllowRedefinitionToAddDeleteMethods' JVM flag.
14:09:23 See https://github.com/reactor/BlockHound/issues/33 for more info.
14:09:23 	at reactor.blockhound.BlockHound$Builder.testInstrumentation(BlockHound.java:435)
14:09:23 	at reactor.blockhound.BlockHound$Builder.install(BlockHound.java:401)
14:09:23 	at reactor.blockhound.BlockHound.install(BlockHound.java:94)
```

https://ci.netty.io/job/netty-centos6-java15-prb/2614/console

It doesn't seem to be related to the updates. The second test job is not available. Could we restart them?
